src/main/scala/wdltool/Main.scala

```diff
@@ -0,0 +1,135 @@
+package wdltool
+
+import java.io.{File => JFile}
+
+import wdl4s.formatter.{AnsiSyntaxHighlighter, HtmlSyntaxHighlighter, SyntaxFormatter}
+import wdl4s.{WdlNamespace, NamespaceWithWorkflow, AstTools}
+import spray.json._
+
+import scala.util.{Failure, Success, Try}
+
+object Main extends App {
+  sealed trait Termination {
+    def output: String
+    def returnCode: Int
+  }
+
+  case class SuccessfulTermination(output: String) extends Termination {
+    override val returnCode = 0
+  }
+
+  case class UnsuccessfulTermination(output: String) extends Termination {
+    override val returnCode = 1
+  }
+
+  case object BadUsageTermination extends Termination {
+    override val returnCode = 1
+    override val output = UsageMessage
+  }
+
+  def dispatchCommand(args: Seq[String]): Termination = {
+    getAction(args) match {
+      case Some(x) if x == Actions.Validate => validate(args.tail)
+      case Some(x) if x == Actions.Highlight => highlight(args.tail)
+      case Some(x) if x == Actions.Inputs => inputs(args.tail)
+      case Some(x) if x == Actions.Parse => parse(args.tail)
+      case _ => BadUsageTermination
+    }
+  }
+
+  def validate(args: Seq[String]): Termination = {
+    continueIf(args.length == 1) {
+      loadWdl(args.head) { _ => SuccessfulTermination("") }
+    }
+  }
+
+  def highlight(args: Seq[String]): Termination = {
+    continueIf(args.length == 2 && Seq("html", "console").contains(args(1))) {
+      loadWdl(args.head) { namespace =>
+        val formatter = new SyntaxFormatter(if (args(1) == "html") HtmlSyntaxHighlighter else AnsiSyntaxHighlighter)
+        SuccessfulTermination(formatter.format(namespace))
+      }
+    }
+  }
+
+  def inputs(args: Seq[String]): Termination = {
+    continueIf(args.length == 1) {
+      loadWdl(args.head) { namespace =>
+        import wdl4s.types.WdlTypeJsonFormatter._
+        val msg = namespace match {
+          case x: NamespaceWithWorkflow => x.workflow.inputs.toJson.prettyPrint
+          case _ => "WDL does not have a local workflow"
+        }
+
+        SuccessfulTermination(msg)
+      }
+    }
+  }
+
+  def parse(args: Seq[String]): Termination = {
+    continueIf(args.length == 1) {
+      SuccessfulTermination(AstTools.getAst(new JFile(args.head)).toPrettyString)
+    }
+  }
+
+  private[this] def continueIf(valid: => Boolean)(block: => Termination): Termination = if (valid) block else BadUsageTermination
```

Comment:
It's a specific type of failed run which happens in multiple places. They could all be UnsuccessfulTermination w/ usage string passed in as the output but to me that doesn't seem to be gaining much.

The termination abstraction helped to significantly debulk the tests, which were doing ridiculous things in cromwell